core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -480,6 +480,34 @@ public static boolean validFileNameExcludingAstrix(String fileName) {
     }
 
     /**
+     * This does a very basic pass at validating that a header's value contains only expected characters according to RFC-5987, and those
+     * that it references.
+     * <p>
+     * https://tools.ietf.org/html/rfc5987
+     * <p>
+     * This is only expected to be used for assertions. The idea is that only readable US-ASCII characters are expected; the rest must be
+     * encoded with percent encoding, which makes checking for a valid character range very simple.
+     *
+     * @return {@code true} if the {@code value} is not obviously wrong.
+     */
+    public static boolean validHeaderValue(String value) {
```

Comment:
Moved to `DeprecationRestChannel` and also added `String requireValidHeader(String value)` that behaves similar to `Objects.requireNonNull` to use for all creations. I did leave it as `public` as it's used by `DeprecationRestHandler` (even though they're in the same package).